ResearchKit/Common/ORKObserver.h

```diff
@@ -0,0 +1,49 @@
+//
+//  ORKObserver.h
+//  ResearchKit
+//
+//  Created by Ricardo Sánchez-Sáez on 22/04/2015.
+//  Copyright (c) 2015 researchkit.org. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@class ORKTaskViewController;
+
+@interface ORKObserver : NSObject
+
+@property (nonatomic, strong) NSArray *keyPaths;
+@property (nonatomic, strong) id target;
+@property (nonatomic) BOOL observing;
+@property (nonatomic) void *context;
+
+@property (nonatomic, weak) id responder;
+@property (nonatomic) SEL action;
+
+- (void)startObserving;
+
+@end
+
+
+@protocol ORKViewControllerObserverProtocol <NSObject>
+@required
+- (void)collectToolbarItemsFromViewController:(UIViewController *)viewController;
+@end
+
+@interface ORKViewControllerObserver : ORKObserver
+
+- (instancetype)initWithTargetViewController:(UIViewController *)target responder:(id <ORKViewControllerObserverProtocol>)responder;
```

Comment:
I wouldn't use the word 'responder'. Either 'delegate' or 'target'. Probably 'delegate'.